api/validator.go

```diff
@@ -0,0 +1,147 @@
+package api
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/collector/types"
+	"github.com/test-network-function/collector/util"
+)
+
+func validatePostRequest(w http.ResponseWriter, r *http.Request) (map[string]interface{}, []string, bool) {
+
+	partnerName := r.FormValue(util.PartnerNameInputName)
+	decodedPassword := r.FormValue(util.DedcodedPasswordInputName)
+	if partnerName == "" || decodedPassword == "" {
+		return nil, nil, false
+	}
+
+	executedBy := r.FormValue(util.ExecutedByInputName)
+
+	if executedBy == "" {
+		util.WriteError(w, "%s", util.ExecutedByMissingErr)
+		return nil, nil, false
+	}
+
+	claimFileMap := parseClaimFile(w, r)
+	if claimFileMap == nil {
+		// error occurred while uploading\converting claim file.
+		return nil, nil, false
+	}
+
+	versions := validateClaimKeys(w, claimFileMap)
+	if versions == nil {
+		return nil, nil, false
+	}
+
+	ocpVersion := versions["ocp"].(string)
+
+	return claimFileMap, []string{partnerName, decodedPassword, executedBy, ocpVersion}, true
+
+}
+
+func validateGetRequest(w http.ResponseWriter, r *http.Request, db *sql.DB) (string, bool) {
+
+	partnerName := r.FormValue(util.PartnerNameInputName)
+	decodedPassword := r.FormValue(util.DedcodedPasswordInputName)
+
+	// If partner name and password are not given return
+	if partnerName == "" && decodedPassword == "" {
```

Comment:
maybe we should change it to or instead of and?

Reply:
Done!